### PR TITLE
#238 EA-1916 implemented tests for search for different body types

### PR DESCRIPTION
**Pull request by rgraf**
Implemented tests for search for different body types

### CODE REVIEW

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+	private AnnotationSearchApiImpl annotationSearchApi;
+	
+	
+	/**
+	 * Create annotations data set before each test execution
+	 * 
+	 * @throws IOException
+	 */
+	@BeforeEach
+	public void createAnnotationSearchApi() throws JsonParseException, IOException {
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: This is not needed to be run before each test. The search client can be initiated once unsing laizy loading in get method

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+		annotationSearchApi = new AnnotationSearchApiImpl();
+	}
+	
+	@Test
+	public void createSemanticTag() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: this is a search not a create test, please rename

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+import eu.europeana.annotation.definitions.model.body.impl.TextBody;
+import eu.europeana.annotation.definitions.model.entity.impl.EdmPlace;
+import eu.europeana.annotation.definitions.model.graph.Graph;
+import eu.europeana.annotation.definitions.model.search.SearchProfiles;
+import eu.europeana.annotation.definitions.model.search.result.AnnotationPage;
+import eu.europeana.annotation.definitions.model.target.Target;
+import eu.europeana.annotation.definitions.model.vocabulary.BodyInternalTypes;
+import eu.europeana.annotation.definitions.model.vocabulary.MotivationTypes;
+
+public class BodyTypesTest extends BaseTaggingTest {
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: This class is used for search tests, please rename it accordingly

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+		Annotation storedAnno = createTag(requestBody);
+		
+		assertTrue(BodyInternalTypes.isSemanticTagBody(storedAnno.getBody().getInternalType()));
+		//validate the reflection of input in output!
+		validateOutputAgainstInput(storedAnno, inputAnno);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_BODY_URI, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET, TOTAL_BY_ID_FOUND);
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: what is the reason to run 3 searches and validate output once? Please include validation after each search

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getHttpUri());
+		assertTrue(target.getHttpUri().equals("http://data.europeana.eu/item/09102/_UEDIN_214"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+
+	@Test
+	public void createGeoTag() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: this is a search not a create test, please rename

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+		// create indexed tag
+		Annotation storedAnno = createTag(requestBody);
+		
+		assertTrue(BodyInternalTypes.isGeoTagBody(storedAnno.getBody().getInternalType()));
+		//validate the reflection of input in output!
+		validateOutputAgainstInput(storedAnno, inputAnno);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET, TOTAL_BY_ID_FOUND);
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: please include validation after each search

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getHttpUri());
+		assertTrue(target.getHttpUri().equals("http://data.europeana.eu/item/09102/_UEDIN_214"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+
+	@Test
+	public void createTag() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: this is a search not a create test, please rename

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+		Annotation storedAnno = createTag(requestBody);
+		
+		assertTrue(BodyInternalTypes.isTagBody(storedAnno.getBody().getInternalType()));
+		//validate the reflection of input in output!
+		validateOutputAgainstInput(storedAnno, inputAnno);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_BODY_VALUE, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET_TAG, TOTAL_BY_ID_FOUND);
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: please validate output after each search

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 9]
```diff
@@ -0,0 +1,354 @@
...
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getHttpUri());
+		assertTrue(target.getHttpUri().equals("http://data.europeana.eu/item/000002/_UEDIN_214"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+	
+	@Test
+	public void createTagText() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: this is a search not a create test, please rename

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 10]
```diff
@@ -0,0 +1,354 @@
...
+		
+		// create indexed tag
+		Annotation storedAnno = createTag(requestBody);
+		
+		assertTrue(BodyInternalTypes.isTextualBody(storedAnno.getBody().getInternalType()));
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_DESCRIBING_BODY_VALUE, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_DESCRIBING_TARGET_SCOPE, TOTAL_BY_ID_FOUND);
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: please include validation after each search

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 11]
```diff
@@ -0,0 +1,354 @@
...
+		assertTrue(target.getSource().equals("http://www.europeana1914-1918.eu/attachments/2020601/20841.235882.full.jpg"));
+		assertNotNull(target.getScope());
+		assertTrue(target.getScope().equals("http://data.europeana.eu/item/07931/diglit_uah_m1"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+	
+	@Test
+	public void createLink() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: this is a search not a create test, please rename

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 12]
```diff
@@ -0,0 +1,354 @@
...
+		
+		String requestBody = getJsonStringInput(LINK_STANDARD);
+		
+		// create indexed tag
+		Annotation storedAnno = createTag(requestBody);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET_LINK, TOTAL_BY_ID_FOUND);
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: please add validation after each search

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 13]
```diff
@@ -0,0 +1,354 @@
...
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getValues());
+		assertTrue(target.getValues().contains("http://data.europeana.eu/item/123/xyz"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+	
+	@Test
+	public void createSemanticLink() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: this is a search not a create test, please rename

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 14]
```diff
@@ -0,0 +1,354 @@
...
+		
+		String requestBody = getJsonStringInput(LINK_SEMANTIC);
+		
+		// create indexed tag
+		Annotation storedAnno = createTag(requestBody);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET_LINK_SEMANTIC, TOTAL_BY_ID_FOUND);
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: please validate output after each search

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 15]
```diff
@@ -0,0 +1,354 @@
...
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getHttpUri());
+		assertTrue(target.getHttpUri().equals("http://data.europeana.eu/item/2059207/data_sounds_T471_5"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+	
+	@Test
+	public void createGraph() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: this is a search not a create test, please rename

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 16]
```diff
@@ -0,0 +1,354 @@
...
+		String requestBody = getJsonStringInput(LINK_SEMANTIC);
+		
+		// create indexed tag
+		Annotation storedAnno = createTag(requestBody);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_BODY_LINK_RESOURCE_URI, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_BODY_LINK_RELATION, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET_LINK_SEMANTIC, TOTAL_BY_ID_FOUND);
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: please add validation after each search

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 17]
```diff
@@ -0,0 +1,354 @@
...
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getHttpUri());
+		assertTrue(target.getHttpUri().equals("http://data.europeana.eu/item/2059207/data_sounds_T471_5"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+	
+	@Test
+	public void createSemanticTagSpecific() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: this is a search not a create test, please rename

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 18]
```diff
@@ -0,0 +1,354 @@
...
+		Annotation storedAnno = createTag(requestBody);
+		
+		assertTrue(BodyInternalTypes.isSemanticTagBody(storedAnno.getBody().getInternalType()));
+		//validate the reflection of input in output!
+		validateOutputAgainstInput(storedAnno, inputAnno);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_BODY_SPECIFIC_RESOURCE, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET, TOTAL_BY_ID_FOUND);
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: please add validation after each search

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 19]
```diff
@@ -0,0 +1,354 @@
...
+		Target target = storedAnno.getTarget();
+		assertNotNull(target.getHttpUri());
+		assertTrue(target.getHttpUri().equals("http://data.europeana.eu/item/09102/_UEDIN_214"));
+
+		// remove tag
+		deleteAnnotation(storedAnno);
+	}
+
+	@Test
+	public void createFullTextResourceTag() throws IOException, JsonParseException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: this is a search not a create test, please rename

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java [hunk 20]
```diff
@@ -0,0 +1,354 @@
...
+		Annotation storedAnno = createTag(requestBody);
+		
+		assertTrue(BodyInternalTypes.isFullTextResourceTagBody(storedAnno.getBody().getInternalType()));
+		//validate the reflection of input in output!
+		validateOutputAgainstInput(storedAnno, inputAnno);
+		
+		// search for indexed id and textual values
+		searchByBodyValue(VALUE_ID+"\""+storedAnno.getAnnotationId().getIdentifier()+"\"", TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_BODY_FULL_TEXT_RESOURCE, TOTAL_BY_ID_FOUND);
+		searchByBodyValue(VALUE_TARGET, TOTAL_BY_ID_FOUND);
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: please add validation after each search

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 1 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: lazy loading implemented

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 2 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: changed to search...

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 3 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: Class renamed to SearchTypesTest

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 4 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: added validations

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 5 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: changed to search...

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 6 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: added validations

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 7 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: changed to search

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 8 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: added validations

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 9 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: changed to search...

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 10 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: added validations

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 11 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: renamed to search

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 12 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: added validations

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 13 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: changed to search...

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 14 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: added validations

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 15 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: changed to search...

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 16 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: added validations

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 17 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: changed to search...

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 18 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: added validations

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 19 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: changed to search...

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java (hunk 20 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/tag/BodyTypesTest.java]: added validations

**Pull request merged**